### ISSUE TITLE
[Improvement] Alternate downloading method for normal and parallel downloads

### ISSUE THREAD

**Issue by Anon-Exploiter** [Akianonymus/gdrive-downloader#15]
Hey demon, it's me ya boy (sauce meme: https://user-images.githubusercontent.com/18597330/99195267-8ab36880-27a6-11eb-86f8-e240f624a881.png) 

So, I actually I was trying to download a folder and it had too much-nested folders, like 10-20 in each folder, and a total of 90 parent directories. 

The thing is, it first completes a directory and then downloads its files when that completes, moves onto other which makes the script rather slow. 

Gupload (from you) on the other hand creates folders first and then uploads files, is it possible to port that feature in this script maybe? 

Would really appreciate using the bandwidth and will save a lot of time I think. 

TL;DR: Creation of directories prior to downloading of files, so, if a user specifies 50 files, it doesn't wait on the directory to complete which let's say has 5 files and move on. It downloads all 50 in parallel.

**Comment by Akianonymus** [comment]
Thanks for the meme reference 😂.

Good suggestion, it surely is.

Now, the problem is, why i wrote gupload that way because we can access the file list locally, just using a find command.

Now, gdrive api doesn't have an endpoint to let me access all the files and folders recursively.

So, i have to it one by one to fetch the folder and sub folder info before anything. Just one drawback is you have to wait till the folders info are fetched.

One more thing i will try to implement, if i can parallelize this process ( fetching folder contents recursively, but not much can be parallelized as we have always to wait for the parent directory info).

**Comment by Anon-Exploiter** [comment]
Haha, thanks. Sounds awesome! 

Maybe introduce a new argument if someone wants to go down the other path (i.e. creation of all directories before it fetches the files)?

**Comment by Akianonymus** [comment]
@Anon-Exploiter Yeah, i forgot to mention, but yes that was implied. Something like --alternate-parallel or whatever.

**Comment by Anon-Exploiter** [comment]
Oh, my bad. Yosh! :rocket:

**Comment by Anon-Exploiter** [comment]
Hey, possible this will be worked on or should I close it?
